### PR TITLE
Patch 4

### PR DESCRIPTION
**Pull request by khong1127**
I got most of the bitstringflicking.md file done now; I just need to add a few sample problems, which I'll come up with real quick later.

### CODE REVIEW

File: markdown/acsl/bitstringflicking.md [hunk 1]
```diff
@@ -1 +1,170 @@
...
+
+<br>
+
+The order of precedence for the operators goes as follows: NOT, SHIFT and CIRC, AND, XOR, and then OR. Continue reading this section for more information on what each of these operators does.
+
+## Bitwise Operators
+
+All of these operators (except *not*) are binary operators, meaning that they must take in two operands, which must be the same length. In the case that they are of different lengths, then zeros will be added to the left of the shorter operand until it is the same length as the other operand. So, for example, if we had the operands `101111` and `101`, 3 zeros would need to be added to *101* to make it `000101`.
+
+Since *not* is a unary operator, this condition does not apply.
```
Comment (Stinggyray) [inline markdown/acsl/bitstringflicking.md:34]: Define unary operator?

File: markdown/acsl/bitstringflicking.md [hunk 2]
```diff
@@ -1 +1,170 @@
...
+
+## Evaluations
+
+This requires knowing all of the bit string operators and how they work. Refer to the Operators section if needed. 
+
+### 1. ((RS–1 (NOT (LC–1 10110))) OR (NOT (LC–1 (RS–1 01001)))) . (This is from North Hollywood ACSL.)
+
+The following solution is broken down into smaller steps to help improve understanding:
+
+Step 0: ((RS–1 (NOT (LC–1 10110))) OR (NOT (LC–1 (RS–1 01001))))
```
Comment (Stinggyray) [inline markdown/acsl/bitstringflicking.md:72]: Make this a numbered list?

Comment (Stinggyray) [reply markdown/acsl/bitstringflicking.md:72]: A table could also work

File: markdown/acsl/bitstringflicking.md [hunk 3]
```diff
@@ -1 +1,170 @@
...
+Step 6: (01001 OR 10111)
+
+Step 7: **11111**
+
+### 2. ((RC-14 (LC-23 01101)) | (LS-1 10011) & (RS-2 10111)) . (This is from the ACSL Wiki.)
+
+Note that the CIRC operation involve circulating by multiple positions beyond one cycle. However, if you think about it, `LC-5 01101` is just `01101`, since one full cycle would have been done. So, accounting for these full cycles, `LC-23 01101` would go through 4 full cycles (since `23 // 5 = 4`); thus, it is equivalent to `LC-3 01101`.
+
+A similar idea applies to `RC-14`. `RC-5` on a 5-bit bit string would just be 1 full cycle that doesn't have any impact on the bit string. So, `RC-14` would go through 2 full cycles (since `14 // 5 = 2`); so, it is equal to `RC-4` (for a 5-bit bit string, at least).
+
```
Comment (Stinggyray) [inline markdown/acsl/bitstringflicking.md:93]: Same deal, numbered list?

File: markdown/acsl/bitstringflicking.md [hunk 4]
```diff
@@ -1 +1,170 @@
...
+Step 3: (LS-1 Cd1Ab) = 01100
+
+Step 4: d1Ab0 = 01100
+
+At this point, you can now solve for a few variables. `d` correlates with `0`. `1` correlates with `1`; notice that while variables are not involved, this is a good way to ensure that you correctly evaluated the bit string. `A` is equal to 1; so, `a` is equal to `0`. `b` is equal to `0`, and the final `0` correlates with 0.
+
+So, altogether, `a` = 0, `b` = 0, and `d` = 0. We don't know about `c` or `e`, but their values don't actually matter. This is because regardless of what they are, the equation will always be reduced to what we have in Step 4. 
+
+Thus, our answer is `x` = `00*0*`, with the `*` standing for any value (`0` or `1`). During the ACSl contest, you should write these solutions out manually (unless noted otherwise) as: `00000`, `00100`, `00001`, and `00101`.
```
Comment (Stinggyray) [inline markdown/acsl/bitstringflicking.md:149]: acsl isn't capitalized properly lol

File: markdown/acsl/bitstringflicking.md [hunk 5]
```diff
@@ -1 +1,170 @@
...
+
+### 4. List all possible values of x (5 bits long) that solve the following equation: (LSHIFT-2 (RCIRC-3 (NOT x))) = 10100 . (This is from someone else's Quizlet.)
+
+Again, `x` will be marked as `abcde`.
```
Comment (Stinggyray) [inline markdown/acsl/bitstringflicking.md:153]: Wording's a bit awkward, maybe "mark x as `abcde`"?

File: markdown/acsl/bitstringflicking.md (hunk 4 above)
Comment (khong1127) [reply markdown/acsl/bitstringflicking.md:149]: oh whoops

**Pull request merged**